### PR TITLE
major updates to python sdk

### PR DESCRIPTION
**Pull request by shabani1**
- Created Indexify client, and classes for Extractor and RepoExtractor (an extractor bound to a repo)
- Updated the getting started page

### CODE REVIEW

File: sdk-py/indexify/repository.py [hunk 1]
```diff
@@ -20,6 +22,22 @@ def list_repositories(base_url: str = DEFAULT_INDEXIFY_URL) -> list[dict]:
     return response.json()['repositories']
 
 
+# TODO: consider tying this back to IndexifyExtractor
+class RepoExtractor:
```
Comment (diptanu) [inline sdk-py/indexify/repository.py:26]: If this is just a data class, we could just use `@dataclass` - would achieve the same outcome with less boilerplate code.

File: sdk-py/indexify/extractor.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+def list_extractors(base_url: str = DEFAULT_INDEXIFY_URL) -> list[dict]:
+    response = requests.get(f"{base_url}/extractors")
+    response.raise_for_status()
+    return response.json()['extractors']
+
+
+# TODO: consider naming this IndexifyExtractor
+class Extractor:
```
Comment (diptanu) [inline sdk-py/indexify/extractor.py:17]: May be use dataclass here unless we are going to add some extra behavior in the form of methods?

File: sdk-py/indexify/repository.py [hunk 3]
```diff
@@ -44,24 +62,15 @@ async def add(self, *chunks: TextChunk) -> None:
                 return await _get_payload(resp)
 
     async def add_documents(self, *documents: dict) -> None:
```
Comment (diptanu) [inline sdk-py/indexify/repository.py:64]: what does this dict going to have? {"text": "....", "metadata": {...} } ? 
It's not quite clear from the signature of the method.

**Review by diptanu — COMMENTED** [review]
Thanks and looks good overall, only have some small comments.

File: src/server.rs [hunk 4]
```diff
@@ -112,6 +112,7 @@ impl Server {
                 post(run_extractors).with_state(repository_endpoint_state.clone()),
             )
             .route(
+                // FIXME: this should be /index/search
```
Comment (diptanu) [inline src/server.rs:115]: Indexes are not global so I wonder if they should be a top level API such as this? for ex, index `foo` could be in one or many repositories.

Comment (shabani1) [reply src/server.rs:115]: Discussed. Diptanu will update paths to read `/repository/search`

File: sdk-py/indexify/repository.py (hunk 3 above)
Comment (shabani1) [reply sdk-py/indexify/repository.py:54]: Removed the extra `add()` method as discussed

File: sdk-py/indexify/extractor.py (hunk 2 above)
Comment (shabani1) [reply sdk-py/indexify/extractor.py:14]: Put a TODO to consider converting to dataclass if no methods

**Pull request merged**